Dockerfile

```diff
@@ -1,64 +1,52 @@
 ENV LANG C.UTF-8
 
-WORKDIR /usr/local
-
-RUN apt update && apt install -y software-properties-common
-RUN apt-add-repository universe && apt update
-RUN apt install -y \
-    build-essential \
-    cmake \
-    git \
-    man \
-    seqtk \
-    time \
-    wget
-
-
-#============================================
-# INSTALL ZLIB
-#============================================
-ENV ZLIB_VERSION 1.2.11
-ENV ZLIB_URL="http://www.zlib.net/zlib-${ZLIB_VERSION}.tar.gz"
-
-RUN wget "$ZLIB_URL" -O - | tar xzf -
-WORKDIR zlib-"$ZLIB_VERSION"
-RUN ./configure --prefix=/usr/ && make && make install
-
-WORKDIR /usr/local
+RUN apt update \
+    && apt install -y software-properties-common \
+    && apt-add-repository universe \
+    && apt update \
+    && apt install --no-install-recommends -y build-essential git cmake wget zlib1g-dev \
+    && rm -rf /var/lib/apt/lists/* \
+    && apt-get clean
 
 #============================================
 # INSTALL BOOST
 #============================================
-ENV BOOST_MAJOR 1
-ENV BOOST_MINOR 62
-ENV BOOST_PATCH 0
-ENV BOOST_URL="https://sourceforge.net/projects/boost/files/boost/${BOOST_MAJOR}.${BOOST_MINOR}.${BOOST_PATCH}/boost_${BOOST_MAJOR}_${BOOST_MINOR}_${BOOST_PATCH}.tar.gz"
-ENV BOOST_LIBS="system,filesystem,iostreams,log,thread,date_time"
-
-RUN wget "$BOOST_URL" -O - | tar xzf -
-WORKDIR "boost_${BOOST_MAJOR}_${BOOST_MINOR}_${BOOST_PATCH}"
-RUN ./bootstrap.sh --prefix=/usr/ --with-libraries="$BOOST_LIBS" && ./b2 install
-
-WORKDIR /usr/local
-
+ENV BM 1
+ENV Bm 62
+ENV BP 0
+ENV BOOST_VERSION "${BM}.${Bm}.${BP}"
+ENV BOOST_V_USCORE "${BM}_${Bm}_${BP}"
+ENV BOOST_URL "http://sourceforge.net/projects/boost/files/boost/${BOOST_VERSION}/boost_${BOOST_V_USCORE}.tar.gz"
+ENV BOOST_LIBS "system,filesystem,iostreams,log,thread,date_time"
+ENV BOOST_DIR "/boost_$BOOST_VERSION"
+RUN mkdir -p "$BOOST_DIR" \
+    && { wget --quiet -O - "${BOOST_URL}" | tar --strip-components=1 -xz -C "${BOOST_DIR}"; } \
+    && apt-get remove -y wget
+WORKDIR "$BOOST_DIR"
+RUN ./bootstrap.sh --prefix=/usr/ --with-libraries="$BOOST_LIBS" \
+    && ./b2 install \
+    && cd .. \
+    && rm -rf "$BOOST_DIR"
 #============================================
 # INSTALL PANDORA
 #============================================
-ENV PANDORA_BRANCH dev
-ENV PANDORA_GIT="https://github.com/rmcolq/pandora.git"
-ENV PANDORA_BUILD RELEASE_WITH_ASSERTS
-
-RUN git clone -b "$PANDORA_BRANCH" --single-branch  --recursive "$PANDORA_GIT"
-WORKDIR pandora
-RUN mkdir -p build
-WORKDIR build
-RUN cmake -DCMAKE_BUILD_TYPE="$PANDORA_BUILD" .. && make
-RUN ln -s $(realpath pandora) /usr/local/bin/pandora
-RUN ctest -V
+# can override the build type with docker's --build-arg command
+# https://docs.docker.com/engine/reference/builder/#arg
+ARG PANDORA_BUILD_TYPE="Release"
+ENV PANDORA_DIR "/pandora/"
+
+COPY . $PANDORA_DIR
+WORKDIR ${PANDORA_DIR}/build
+RUN cmake -DCMAKE_BUILD_TYPE="$PANDORA_BUILD_TYPE" .. \
+    && make -j4 \
+#    && ctest -V \
```

Comment:
The issue here is that tests don't pass on release mode because of our excessive use of assert statements. 
We shouldn't really be distributing a non-release version of pandora IMO. So, to distrubute a release version, we have to skip tests.
The assert statements cause a sizeable hit to performance, as we've sen before. 
Bear in mind that we now run tests in the CI, so we will know if a commit's tests don't pass. 
I think we should wait until #166 is fixed before switching this back on.
BUT, I left it in there, commented-out, so that when you build locally, you can obviously uncomment it and change the build type on the CLI if you wish.

Reply:
Yeah, I commented about this just in the comment above. In summary, I think we should favour a binary that we are sure is behaving as we expect, i.e. compiled with `RELEASE_WITH_ASSERTS` and all tests passing, than a faster binary. You are right that the correct solution is solving https://github.com/rmcolq/pandora/issues/166, so I'm proposing we produce a slower binary for now, and we keep the tests to ensure all is fine, and solve https://github.com/rmcolq/pandora/issues/166 ASAP (i.e. in this or in the next PR)